ANMN/LTSP/TSaggregator/test/geoserverCatalog.py

```diff
@@ -0,0 +1,98 @@
+## read all ANMN/ABOS files names form the geoserver
+"""
+geoserverCatalog.py
+
+Collect files names from the AODN geoserver according to several conditions
+Output a list of urls and optionally write into a text file
+
+usage: geoserverCatalog.py [-h] -var VAR -site SITE [-ft FEATURETYPE] [-fv FV]
+                           [-realtime REALTIME] [-ts TIMESTART] [-te TIMEEND]
+                           [-out OUTFILELIST]
+
+optional arguments:
+  -h, --help          show this help message and exit
+  -var VAR            name of the variable of interest, like TEMP
+  -site SITE          site code, like NRMMAI
+  -ft FEATURETYPE     feature type, default timeseries
+  -fv FV              file version, 0, 1 or 2 only
+  -realtime REALTIME  real time data wanted? True or False. Default False
+  -ts TIMESTART       start time like 2015-12-01. Default 1944-10-15
+  -te TIMEEND         end time like 2018-06-30. Default today's date
+  -out OUTFILELIST    name of the file to store the selected files urls.
+                      Default: fileList.csv
+"""
+
+
+import sys
+import argparse
+from datetime import datetime, timedelta
+from distutils.util import strtobool
+
+from owslib.wfs import WebFeatureService
+from owslib.fes import *
+from owslib.etree import etree
+import pandas as pd
+
+
+
+def args():
+    parser = argparse.ArgumentParser(description="Get a list of urls from the AODN geoserver")
+    parser.add_argument('-var', dest='varname', help='name of the variable of interest, like TEMP', required=True)
+    parser.add_argument('-site', dest='site', help='site code, like NRMMAI',  required=True)
+    parser.add_argument('-ft', dest='featuretype', help='feature type, default timeseries', default='timeseries', required=False)
+    parser.add_argument('-fv', dest='fileversion', help='file version 1 or 2 only', default=1, type=int, required=False)
+    parser.add_argument('-ts', dest='timestart', help='start time like 2015-12-01. Default 1944-10-15', default='1944-10-15', required=False)
+    parser.add_argument('-te', dest='timeend', help='end time like 2018-06-30. Default today\'s date', default=str(datetime.now())[:10], required=False)
+    parser.add_argument('-out', dest='outFileList', help='name of the file to store the selected files urls. Default: filelist.csv', default="", required=False)
+    parser.add_argument('-realtime', dest='realtime', help='real time data wanted?', required=False)
+
+    vargs = parser.parse_args()
+    return(vargs)
+
+
+def get_urls(varname, site, featuretype='timeseries', fileversion=1, realtime='False', timestart='1944-10-15', timeend=str(datetime.now())[:10], outFileList=''):
```

Comment:
Another thought on this function, since it is hard-coded to query the "moorings_all_map" layer, and it wouldn't make sense to generalise it further,  the  function could be called `get_moorings_urls`.

Reply:
good point. function name changed as suggested.